### PR TITLE
Refactor in java bytecode instrument [TG-2331]

### PR DESCRIPTION
**Pull request by romainbrenguier**
This PR contains 5 small independent code improvements in java_bytecode_instrument.

### CODE REVIEW

File: src/java_bytecode/java_bytecode_instrument.cpp [hunk 1]
```diff
@@ -73,29 +69,26 @@ class java_bytecode_instrumentt:public messaget
...
-const std::vector<irep_idt> exception_needed_classes =
-{
+const std::vector<std::string> exception_needed_classes = { // NOLINT
   "java.lang.ArithmeticException",
   "java.lang.ArrayIndexOutOfBoundsException",
   "java.lang.ClassCastException",
   "java.lang.NegativeArraySizeException",
-  "java.lang.NullPointerException"
-};
+  "java.lang.NullPointerException"};
```
Comment (smowton) [inline src/java_bytecode/java_bytecode_instrument.cpp:83]: The existing brace style seems more consistent with our usual style, but I guess clang-format wouldn't take it?

**Review by smowton — APPROVED** [review]

File: src/java_bytecode/java_bytecode_instrument.cpp (hunk 1 above)
Comment (romainbrenguier) [reply src/java_bytecode/java_bytecode_instrument.cpp:83]: clang-format puts it that way, but it's one of those cases where clang-format and cpp-lint disagree

**Review by thk123 — APPROVED** [review]
Nice.

**Comment by thk123** [comment]
Would of course be good to see a test-gen pointer bump

File: src/java_bytecode/java_bytecode_instrument.cpp [hunk 2]
```diff
@@ -533,7 +522,7 @@ codet java_bytecode_instrumentt::instrument_expr(
     }
     else if(statement==ID_java_new_array)
     {
-      // this correpond to new array so we check that
+      // this correponds to new array so we check that
```
Comment (peterschrammel) [inline src/java_bytecode/java_bytecode_instrument.cpp]: corresponds

**Pull request merged**